### DISCUSSION TITLE
How to set up a 2-NIC, ltsp19 testing env. in Virtualbox

### DISCUSSION THREAD

**Discussion by dgroos** [ltsp/ltsp#264] [Q&A]
Setting up a testing environment for ltsp19 on Ubuntu 18.04. This thread can serve as a public guide where the setup can be refined and adapted to different needs. Perhaps create a wiki page at some point when ready.

1. Downloaded and installed Virtualbox 6.0.12 along with the VB guest addons. 

1. Added a VM. Necessary setting: a single network adapter using NAT. This allowed internet access on Adapter 1.

1. Launched and installed Ubuntu 18.04.03, Desktop 64 bit. Added a few useful utilities.

1. Shut down, added another network adapter to this VM (selected "Adapter 2 and checked the "enable network adapter) and attached it to "Internal Network (NOT bridged otherwise it will offer pxe booting to the WAN)". Started the VM and set this new NIC to have IPv4 set to Manual with Address to 192.186.67.1, Netmask to 255.255.255.0 and Gateway to 0.0.0.0.

1. Followed [the updated directions](https://ltsp.github.io/docs/installation/) using `ltsp dnsmasq --proxy-dhcp=0` for network setup. To create the image used `ltsp image /` as I'm using chrootless (ltsp-pnp) approach. 

1. Added another VM to serve as the client. This one I set to not have a hard drive and to boot from the network. Set a single network adapter to "Internal Network (NOT bridged otherwise it will attempt to boot from the WAN)".. Booted it and... it worked (!). Tested internet access on the client and that worked.

Any suggested changes to this setup? To this order of operations?

Thanks,
David

**Comment by alkisg** [comment]
Do not use bridged with 192.168.67.1 because it will give IPs to the other clients on the network. Use internal network.

**Comment by dgroos** [comment]
I updated the above instructions based on the advise and successfully accomplishing the testing goal.

Thanks for the advise @alkisg !
David

**Comment by alkisg** [comment]
David it might be a good idea to turn this into a wiki page:

https://github.com/ltsp/community/wiki

**Comment by dgroos** [comment]
Will do this weekend, thanks for the idea/link, Alkis.
David

**Comment by dgroos** [comment]
"this" weekend has apparently arrived :-P I'll remove the "Under development sign" if it no one sees anything really wrong...
Thanks!
https://github.com/ltsp/community/wiki/Create-an-LTSP19-testing-environment-in-VirtualBox

**Comment by dgroos** [comment]
no concerns noted w/it, closing...
